libraries/AP_Notify/Display.cpp

```diff
@@ -297,11 +297,22 @@ bool Display::init(void)
     // initialise driver
     switch (pNotify->_display_type) {
         case DISPLAY_SSD1306:
-            _driver = new Display_SSD1306_I2C(hal.i2c_mgr->get_device(NOTIFY_DISPLAY_I2C_BUS, NOTIFY_DISPLAY_I2C_ADDR));
+            for(uint8_t i=0; i<3; i++) {
+                _driver = new Display_SSD1306_I2C(hal.i2c_mgr->get_device(i, NOTIFY_DISPLAY_I2C_ADDR));
+                if (_driver && _driver->hw_init()) {
```

Comment:
If *hw_init* is a good detection mechanism (I have my doubts about that, but @tridge has probably a more trustable opinion), then I'd rather have a probe method like is done other libraries - then only the class can create and destroy an object of its type.

In any case, *hw_init* is called again in line 323, that should be removed.